packages/events-api/src/adapter.ts

```diff
@@ -0,0 +1,156 @@
+import EventEmitter from 'events'; // tslint:disable-line
+import http, { IncomingMessage, ServerResponse } from 'http';
+import debugFactory from 'debug'; // tslint:disable-line
+import { createHTTPHandler } from './http-handler';
+
+const debug = debugFactory('@slack/events-api:adapter');
+
+/**
+ * An adapter for Slack's Events API.
+ */
+export class SlackEventAdapter extends EventEmitter {
+  /**
+   * The token used to authenticate signed requests from Slack's Events API.
+   */
+  public readonly signingSecret: string;
+
+  /**
+   * Whether to include the API event bodies in adapter event consumers.
+   */
+  public includeBody: boolean;
+
+  /**
+   * Whether to include request headers in adapter event consumers.
+   */
+  public includeHeaders: boolean;
+
+  /**
+   * When `true`, prevents the adapter from responding by itself and leaves that up to consumers.
+   */
+  public waitForResponse: boolean;
+
+  /**
+   * The HTTP server this adapter might be running, created in {@link start}.
+   */
+  private server?: http.Server;
+
+  /**
+   * @param signingSecret - The token used to authenticate signed requests from Slack's Events API.
+   * @param opts.includeBody - TODO:
+   * @param opts.includeHeaders - TODO:
+   * @param opts.waitForResponse - TODO:
+   */
+  constructor(
+    signingSecret: string | String,
+    {
+      includeBody = false,
+      includeHeaders = false,
+      waitForResponse = false,
+    }: EventAdapterOptions = {},
+  ) {
+    if (typeof signingSecret !== 'string' && !(signingSecret instanceof String)) {
+      throw new TypeError('SlackEventAdapter needs a signing secret');
+    }
+
+    super();
+
+    this.signingSecret = signingSecret as string;
+    this.includeBody = includeBody;
+    this.includeHeaders = includeHeaders;
+    this.waitForResponse = waitForResponse;
+
+    debug('adapter instantiated - options: %o', {
+      includeBody,
+      includeHeaders,
+      waitForResponse,
+    });
+  }
+
+  /**
+   * Creates an HTTP server to listen for event payloads.
+   *
+   * @param path - (UNUSED) The path to listen on.
+   */
+  public createServer(path: string = '/slack/events'): Promise<http.Server> {
+    // TODO: options (like https)
+    // NOTE: this is a workaround for a shortcoming of the System.import() tranform
```

Comment:
I think it should definitely be replaced with something else, as seeing `Promise.resolve().then(() => { ... })` is definitely a pattern that should be explained. I don't know what to write here, though (because I largely don't understand why it's like this 🤷‍♀)